catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswQueryFactory.java

```diff
@@ -0,0 +1,340 @@
+
+        if (filter == null) {
+            throw new CswException("Invalid Filter Expression",
+                    CswConstants.NO_APPLICABLE_CODE,
+                    null);
+        }
+        visitor.setVisitedFilter(filter);
+        if (typeNames.contains(new QName(CswConstants.CSW_OUTPUT_SCHEMA,
+                CswConstants.CSW_RECORD_LOCAL_NAME,
+                CswConstants.CSW_NAMESPACE_PREFIX))) {
+
+            try {
+                visitor.setVisitedFilter((Filter) filter.accept(visitor, null));
+            } catch (UnsupportedOperationException ose) {
+                throw new CswException(ose.getMessage(),
+                        CswConstants.INVALID_PARAMETER_VALUE,
+                        null);
+            }
+        }
+        return visitor;
+    }
+
+    private SortBy buildSort(SortByType sort) throws CswException {
+        if (sort == null || sort.getSortProperty() == null) {
+            return null;
+        }
+
+        SortBy[] sortByArr = parseSortBy(sort);
+
+        if (sortByArr.length > 1) {
+            LOGGER.warn("Query request has multiple sort criteria, only primary will be used");
+        }
+
+        SortBy sortBy = sortByArr[0];
+
+        if (sortBy.getPropertyName() == null) {
+            LOGGER.warn("No property name in primary sort criteria");
+            return null;
+        }
+
+        if (!DefaultCswRecordMap.getDefaultCswRecordMap()
+                .hasDefaultMetacardFieldForPrefixedString(sortBy.getPropertyName()
+                                .getPropertyName(),
+                        sortBy.getPropertyName()
+                                .getNamespaceContext())) {
+            throw new CswException("Property " + sortBy.getPropertyName()
+                    .getPropertyName() + " is not a valid SortBy Field",
+                    CswConstants.INVALID_PARAMETER_VALUE,
+                    "SortProperty");
+        }
+
+        String name = DefaultCswRecordMap.getDefaultCswRecordMap()
+                .getDefaultMetacardFieldForPrefixedString(sortBy.getPropertyName()
+                                .getPropertyName(),
+                        sortBy.getPropertyName()
+                                .getNamespaceContext());
+
+        PropertyName propName = new AttributeExpressionImpl(new NameImpl(name));
+
+        return new SortByImpl(propName, sortBy.getSortOrder());
+    }
+
+    private SortBy[] parseSortBy(SortByType sortByType) throws CswException {
+        JAXBElement<SortByType> sortByElement =
+                new net.opengis.filter.v_1_1_0.ObjectFactory().createSortBy(sortByType);
+
+        return (SortBy[]) parseJaxB(sortByElement);
+    }
+
+    private Object parseJaxB(JAXBElement<?> element) throws CswException {
+        Parser parser = new Parser(PARSER_CONFIG);
+        InputStream inputStream = null;
+
+        try {
+            inputStream = marshalJaxB(element);
+            return parser.parse(inputStream);
+        } catch (JAXBException e) {
```

Comment:
Can't we use `catch (JAXBException | IOException | ...)` here, not add the caught exception name or message to the exception message and simply pass the exception caught as the cause in the `CswException`?

If we want or need to pass in the caught exception's class name and message, can't we just do `throw new CswException(String.format("Failed to parse Element (%s): %s", e.getClass().getSimpleName(), e.getMessage())`?